packages/components/src/mobile/keyboard-aware-flat-list/index.ios.js

```diff
@@ -19,53 +26,64 @@ export const KeyboardAwareFlatList = ( {
 	autoScroll,
 	scrollViewStyle,
 	inputAccessoryViewHeight,
+	onScroll,
 	...listProps
-} ) => (
-	<KeyboardAwareScrollView
-		style={ [ { flex: 1 }, scrollViewStyle ] }
-		keyboardDismissMode="none"
-		enableResetScrollToCoords={ false }
-		keyboardShouldPersistTaps="handled"
-		extraScrollHeight={ extraScrollHeight }
-		extraHeight={ 0 }
-		inputAccessoryViewHeight={ inputAccessoryViewHeight }
-		enableAutomaticScroll={ autoScroll === undefined ? false : autoScroll }
-		ref={ ( ref ) => {
-			this.scrollViewRef = ref;
-			innerRef( ref );
-		} }
-		onKeyboardWillHide={ () => {
-			this.keyboardWillShowIndicator = false;
-		} }
-		onKeyboardDidHide={ () => {
-			setTimeout( () => {
-				if (
-					! this.keyboardWillShowIndicator &&
-					this.latestContentOffsetY !== undefined &&
-					! shouldPreventAutomaticScroll()
-				) {
-					// Reset the content position if keyboard is still closed.
-					if ( this.scrollViewRef ) {
-						this.scrollViewRef.scrollToPosition(
-							0,
-							this.latestContentOffsetY,
-							true
-						);
+} ) => {
+	const latestContentOffsetY = useSharedValue( -1 );
+	const scrollHandler = useAnimatedScrollHandler( {
+		onScroll: ( event ) => {
+			const { contentOffset } = event;
+			latestContentOffsetY.value = contentOffset.y;
+			onScroll( event );
+		},
+	} );
+	return (
+		<AnimatedKeyboardAwareScrollView
+			style={ [ { flex: 1 }, scrollViewStyle ] }
+			keyboardDismissMode="none"
+			enableResetScrollToCoords={ false }
+			keyboardShouldPersistTaps="handled"
+			extraScrollHeight={ extraScrollHeight }
+			extraHeight={ 0 }
+			inputAccessoryViewHeight={ inputAccessoryViewHeight }
+			enableAutomaticScroll={
+				autoScroll === undefined ? false : autoScroll
+			}
+			ref={ ( ref ) => {
+				this.scrollViewRef = ref;
+				innerRef( ref );
+			} }
+			onKeyboardWillHide={ () => {
+				this.keyboardWillShowIndicator = false;
+			} }
+			onKeyboardDidHide={ () => {
```

Comment:
Not relevant for this PR but we should update this code to avoid using that many inline functions 😅

Reply:
I replaced the inline functions in https://github.com/WordPress/gutenberg/pull/39617/commits/842b09b296409c99705ed5a09972ce81da02a4f5.